Brenosalv/feature/updated-menu-design

Brenosalv opened the pull request:
#303 
#314 
#324 

## Changes

- [x] Add redesigned product menu;
- [x] Add redesigned resources menu

## Tests / Screenshots

-   ![image](https://github.com/estuary/marketing-site/assets/60396753/eab3b6da-54e7-4f5c-a64c-ca0b145d7d04)
-   ![image](https://github.com/estuary/marketing-site/assets/60396753/f3699b45-b147-4490-863e-7a1f7414fc4e)

Brenosalv commented on package.json:
```diff
@@ -106,6 +106,7 @@
     "lint:fix": "eslint . --ext .js,.jsx,.ts,.tsx --fix",
     "format": "prettier --check \"**/*.{js,jsx,ts,tsx,json,md}\"",
     "format:fix": "prettier --write \"**/*.{js,jsx,ts,tsx,json,md}\"",
+    "fix": "yarn lint:fix && yarn format:fix",
```
@travjenkins I added this script to make fixing things easier before committing, since we don't have the fix in husky.

travjenkins replied:
Makes sense

travjenkins commented on src/components/HeaderNavbar/CardItem/styles.ts:
```diff
@@ -0,0 +1,82 @@
+    font-size: 0.875rem;
+    font-weight: 400;
+    line-height: 30px;
+    color: #ffffff;
```
Could we move the color into a parent to reduce duplication. I know this is a tiny thing but trying to keep CSS as small as possible.

Brenosalv replied:
I can do this but it's a problem that is present in all the css codes. We could create an issue to store all the colors of the entire application in a file as ```globalStylies.ts``` and import from there.

travjenkins replied:
I would like to reduce how much we're importing and reapplying styles if we can. I would like to style a lot of stuff at the top level and then only use styling to override when needed.

travjenkins replied:
And yeah - this is a big problem in the app for sure... this is not coming from the code of just this PR :D

travjenkins commented on src/components/HeaderNavbar/Resources/index.tsx:
```diff
@@ -65,35 +66,34 @@ const HeaderNavbarResources = ({ active, setActive }) => {
                             target="_blank"
-                            href={webinarsUrl}
-                            className="cta-button"
+                            href={webinarsUrl} // TODO: replace this URL with the tour video URL
```
FYI - a `todo` is still in here.

Brenosalv replied:
Yeah, I was keeping it to remove when working on the carousel.

travjenkins commented on src/globalStyles.ts:
```diff
@@ -70,13 +74,30 @@ export const BaseButtonFilledStyling = `
+                background-color: transparent;
+                border: 2px solid #5072eb;
+            `;
+        default:
+            return lightStyles;
```
I think we're fine just having `light` be the default and not checked up above. That way we don't need to store off the lightStyles. 

Also, can we make the border styling shared?

Brenosalv replied:
Makes sense. Will do this.

travjenkins commented on src/images/product-tour-2min.png:
What changed with this?

Brenosalv replied:
I replaced the image with a bigger one.

travjenkins replied:
Cool - asking cause I was thinking "I have seen this exact image before" lol

travjenkins reviewed (changes requested):
a few small changes

travjenkins commented on src/components/HeaderNavbar/Card/styles.ts:
```diff
@@ -5,17 +5,27 @@ interface ContainerProps {
+    ${BaseMenuStyling};
+    width: 100vw;
+    height: 100vh;
+    background-color: #00000099;
```
This looks wrong... there are more than 6 digits. Does hex support opacity with the last two digits?

travjenkins commented on src/components/header.tsx:
```diff
@@ -147,71 +147,73 @@ const Header = (props: { fixedHeader?: boolean }) => {
+                        <button
+                            onClick={() => setMobileMenuOpen((open) => !open)}
+                            className="global-header-mobile-menu-button"
+                            title="Menu"
```
Can we update this to `Navigation Menu` to be more specific.

Brenosalv replied:
Sure, makes sense.

travjenkins commented on src/components/HeaderNavbar/Resources/items.tsx:
```diff
@@ -1,36 +1,63 @@
-        to: 'https://www.youtube.com/watch?v=Ys5BoNqKljc',
+        hasChevronIcon: true,
+        Image: () => (
+            <StaticImage src="../../../svgs/microphone.svg" alt="About us" />
```
Can we try to get some better `alt` text in here?

travjenkins replied:
Same applies to the two images down below.

Brenosalv replied:
Okay, I will follow the same idea of the About us from the image up above. This way the alt text would be "Podcasts", "Webinars" and "Youtube".

Brenosalv commented on src/components/Carousel/index.tsx:
```diff
@@ -0,0 +1,167 @@
+    }, []);
+
+    React.useEffect(() => {
+        if (hasMounted.current && slideRefs.current[currentSlide]) {
```
@travjenkins The scrolling problem here was caused by the ```scrollIntoView()``` being called on first render. Now we are preventing the component to call it initially using a ref.

Brenosalv commented:
@travjenkins Also, I removed the ```keen-slider``` package and replaced the connectors page carousel with the new one allowing the same colors to the dots.

Brenosalv commented on src/components/Carousel/styles.ts:
```diff
@@ -0,0 +1,84 @@
+
+export const Viewport = styled.ul`
+    display: flex;
+    align-items: center;
```
https://github.com/estuary/marketing-site/issues/314https://github.com/estuary/marketing-site/issues/314

I resolved the issue by centering the slide content. I believe this is the maximum reduction we can achieve for the spacing between the testimonials and dots from the carousel.

travjenkins replied:
I think this is okay for now. It is better than before in my opinion. Probably see what Lucas thinks.

Eventually we should put some effort into adjusting the height dynamically.

Brenosalv replied:
I thought about adjusting the height dinamically but I personally didn't like the UX like that.

I talked to Lucas and had a positive feedback about the spacing.

Also, I adjusted the homepage testimonials' logo and font sizes according to figma design. They were too big for mobile.

travjenkins commented on src/components/HeaderNavbar/Card/index.tsx:
```diff
@@ -1,14 +1,28 @@
+    const isSmallScreen = useMediaQuery('(max-width:1024px)');
+    const isMobile = isRealMobile || isSmallScreen;
+
+    if (isMobile) {
+        return (
+            <Wrapper ref={customRef} {...props}>
+                <Divider />
+                <Container>{children}</Container>
+            </Wrapper>
+        );
+    }
```
What is driving the need of this? Are we not able to style `MenuBackground` for small screens with just css?

Brenosalv replied:
You're totally right, there is no need of this. Changed to css only.

Brenosalv commented:
@travjenkins As we don't have a mobile version of the redesigned menus, I felt free to develop the responsiveness including an accordion when it's in mobile devices. Also, the mobile version in prod is bugged when clicking on the header links "Product" and "Resources" requiring us sometimes double clicking to open them. Please let me know your thoughts.

travjenkins commented:
Only major feedback on mobile menu is I think the items should be spaced vertically a bit more

Current
![image](https://github.com/estuary/marketing-site/assets/270078/e89d30d5-4d32-49d0-bad4-b6a7de29ce79)

Increasing `gap` to `12px`
![image](https://github.com/estuary/marketing-site/assets/270078/2c563ff8-be83-48b0-ab21-f4db0a6c6291)

travjenkins reviewed (dismissed):
lgtm

travjenkins reviewed (dismissed):
lgtm

travjenkins reviewed (dismissed):
lgtm - this **really** makes the nav look better.

travjenkins reviewed (approved):
lgtm - hit the preview app and looked around and looked at the changes in the styling (post merge)
